code/Items/Data/ItemRegistry.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace ssl.Items.Data
+{
+	/// <summary>
+	/// Stores all references of item archetypes.
+	/// </summary>
+	public class ItemRegistry
+	{
+		private readonly Dictionary<string, Item> registry;
+
+		public ItemRegistry()
+		{
+			registry = new Dictionary<string, Item>();
+			
+			RegisterItems();
+		}
+
+		public Item Apple => registry["item.food.apple"]; //Item shortcut example
+		
+		public void Add(Item item)
+		{
+			registry.Add(item.Id, item);
+		}
+
+		public Item GetItemById(string id)
+		{
+			if (!registry.ContainsKey(id))
+			{
+				throw new NotImplementedException( "This item does not exist.");
+			}
+
+			return registry[id];
+		}
+
+		/// <summary>
+		/// Register all the game items in the registry.
+		/// In the future, this should disappear and should load items from an XML / JSON file.
+		/// </summary>
+		private void RegisterItems()
+		{
+			Add(new ItemFood("item.food.apple", "Apple", "", 1, 10)); //Item registration Example
```

Comment:
Juste une préocupation. item.food est commum a tous les ItemFood tu penses qu'on pourrais set le début de l'id dans les parents ?
Dans Item avoir le "item" et dans le ItemFood avoir "item.food" etc ...

Reply:
Ouais c'est ce que je me demandais ce serait génial. Je peux le faire si tu veux je me suis juste demandé si c'était intuitif vu que ça se trouve quelqu'un va mettre sans faire gaffe l'id complet (item.food.apple) alors qu'on a seulement besoin de apple.
Je peux aussi essayer de gérer ce cas, tu me dis juste si tu veux que je le fasse et ça part

Reply:
Au pire osef for now on verra quand on aura 20 items